### DISCUSSION TITLE
Query string not liking values with & sign.

### DISCUSSION THREAD

**Discussion by trev33** [livewire/livewire#4072] [7. Bugs (archived)]
Laravel v8.66
Livewire v2.7.2

**Steps to reproduce:**

1. Base Install for Laravel and Livewire components.
2. php artisan make:livewire search-post
3. Add livewire component to a page @livewire('search-post')
4. Type a search with & sign. ex: _Art & Animation_
5. The results will have a query string as: localhost:8000/?search=Art+&+Animation&Animatio=&Animati=&Animat=&Anima=&Anim=&Ani=&An=&A=
6. Refresh page with URL Chrome will change it to localhost:8000/?search=Art losing everything after the & sign.

**SearchPost.php**
```
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class SearchPost extends Component
{
    public $search;

    protected $queryString = ['search'];

    public function render()
    {
        return view('livewire.search-post');
    }
}

```
**search-post.blade.php**
```
<div>
    <input wire:model="search" type="search" placeholder="Search posts by title...">
</div>
```

**Video of what is happening**
livewire v2.7.2 you can see it doesn't like the & sign

https://user-images.githubusercontent.com/43149308/138475637-c975e0f6-1edf-44a8-aab2-d4e86844e462.mp4

**Downgrading I found Livewire v2.5.5 it was totally fine so it started happening in v2.6.0**
Video in 2.5.5 working fine

https://user-images.githubusercontent.com/43149308/138489272-98f9b572-82b3-46d7-a828-d7a0ae870323.mp4

I've been enjoying Livewire, keep up the outstanding work!

Edit: Added video, and when it was fine.

**Comment by joshhanley** [comment]
@trev33 ah that's interesting, I thought it would encode the "&".

Maybe it's not encoding the data for the query string properly!

**Reply by trev33** [reply]
Yes, it should encode the "&" to %26 but it doesn't.

In version [2.6.0](https://github.com/livewire/livewire/releases/tag/v2.6.0) that this started to happen, I only see two changes with query strings #3685 that you did @joshhanley and the other #3524 that @nullthoughts made. I see on line [151 of SyncBrowserHistory.js](https://github.com/livewire/livewire/blob/54864194f05daa62837ec9200e7c3cbbeb90b3ce/js/component/SyncBrowserHistory.js#L151) it's using decodeURIComponent. If instead we use decodeURL it cannot decode characters (& # , ? / $ : @ = +). Where decodeURIComponent can decode those. 

If I go into that new Laravel / Livewire project created and search for "& # , ? / $ : @ = +" the results are:
http://localhost:8000/?search=&+#+,+?+/+$+:+@+=++

If I make the changes to use **decodeURL** on [151 of SyncBrowserHistory.js](https://github.com/livewire/livewire/blob/54864194f05daa62837ec9200e7c3cbbeb90b3ce/js/component/SyncBrowserHistory.js#L151) and search for "& # , ? / $ : @ = +" the results are:
http://localhost:8000/?search=%26+%23+%2C+%3F+%2F+%24+%3A+%40+%3D+%2B

Adding a new text field called filter and typing "Art & Animation" and for search field type "& # , ? / $ : @ = +" the results are:
http://localhost:8000/?search=%26+%23+%2C+%3F+%2F+%24+%3A+%40+%3D+%2B&filter=Art+%26+Animation

Seems to have fixed the problem. Might be a possible fix or help you guys fix this.

**Reply by joshhanley** [reply]
@trev33 yep makes sense, I don't think anything in my PR would have changed the encoding.

Thanks for testing that out!

@nullthoughts do you have any thoughts on making the above changes to `decodeURL`?

**Reply by nullthoughts** [reply]
@trev33 @joshhanley Looking into this. I vaguely remember there being a reason for using `decodeURIComponent` instead of `decodeURL` (and it possibly being something to do with the handling of ampersand characters).

**Reply by nullthoughts** [reply]
@trev33 Looks like you are correct, `decodeURI()` fixes the issue at hand. Finally got around to spending some time with it today, thought I was going nowhere until I realized the new test I had added was incorrect (you might have seen a prior comment come through, ignore it). I also appended to the tests for this scenario and all is passing.

Do you want to open a PR for it? Or if it's easier, I have the branch/code ready to go

**Reply by joshhanley** [reply]
@nullthoughts thanks for looking into it, much appreciated! 🙂

If you have the code ready to go and @trev33 hasn't already also got code ready to go, then I'd say submit the PR.

Thanks!

**Reply by nullthoughts** [reply]
Thanks @joshhanley , PR #4155 submitted
